homeassistant/components/philips_js/binary_sensor.py

```diff
@@ -0,0 +1,109 @@
+"""Philips TV binary sensors."""
+from __future__ import annotations
+
+from dataclasses import dataclass
+
+from haphilipsjs import PhilipsTV
+
+from homeassistant.components.binary_sensor import (
+    BinarySensorEntity,
+    BinarySensorEntityDescription,
+)
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.core import HomeAssistant, callback
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+from homeassistant.helpers.update_coordinator import CoordinatorEntity
+
+from . import PhilipsTVDataUpdateCoordinator
+from .const import DOMAIN
+
+
+@dataclass
+class PhilipsTVBinarySensorEntityDescription(BinarySensorEntityDescription):
+    """A entity description for Philips TV binary sensor."""
+
+    def __init__(self, recording_entry, recording_value, *args, **kwargs) -> None:
+        """Set up a binary sensor entity description and add additional attributes."""
+        super().__init__(*args, **kwargs)
+        self.recording_entry: str = recording_entry
```

Comment:
```suggestion
```

I think we can remove it, as it is always the same, and IMO if there will be added a different type, the whole code needs refactoring as some values are hardcoded:
- [`coordinator.api.json_feature_supported("recordings", "List")`](https://github.com/Floyer007/core/blob/philips_js-recording-feature/homeassistant/components/philips_js/binary_sensor.py#L63)
- [`for rec in api.recordings_list["recordings"]:`](https://github.com/Floyer007/core/blob/philips_js-recording-feature/homeassistant/components/philips_js/binary_sensor.py#L74)

Reply:
That's true. Checking for "RecError" might need another logic